DNN Platform/Library/Entities/Users/Profile/UserProfile.cs

```diff
@@ -3,688 +3,710 @@
-        /// Initialises the Profile with an empty collection of profile properties.
-        /// </summary>
-        /// <remarks></remarks>
-        /// <param name="portalId">The name of the property to retrieve.</param>
-        /// <param name="useDefaults">A flag that indicates whether the profile default values should be
-        /// copied to the Profile.</param>
-        /// -----------------------------------------------------------------------------
-        public void InitialiseProfile(int portalId, bool useDefaults)
-        {
-            this._profileProperties = ProfileController.GetPropertyDefinitionsByPortal(portalId, true, false);
-            if (useDefaults)
-            {
-                foreach (ProfilePropertyDefinition ProfileProperty in this._profileProperties)
-                {
-                    ProfileProperty.PropertyValue = ProfileProperty.DefaultValue;
-                }
-            }
-        }
-
-        /// -----------------------------------------------------------------------------
-        /// <summary>
-        /// Sets a Profile Property Value in the Profile.
-        /// </summary>
-        /// <remarks></remarks>
-        /// <param name="propName">The name of the propoerty to set.</param>
-        /// <param name="propValue">The value of the propoerty to set.</param>
-        /// -----------------------------------------------------------------------------
-        public void SetProfileProperty(string propName, string propValue)
-        {
-            ProfilePropertyDefinition profileProp = this.GetProperty(propName);
-            if (profileProp != null)
-            {
-                profileProp.PropertyValue = propValue;
-
-                // Set the IsDirty flag
-                if (profileProp.IsDirty)
-                {
-                    this._IsDirty = true;
-                }
-            }
-        }
-
-        private string GetListValue(string listName, string value)
-        {
-            ListController lc = new ListController();
-            int entryId;
-            if (int.TryParse(value, out entryId))
-            {
-                ListEntryInfo item = lc.GetListEntryInfo(listName, entryId);
-                if (item != null)
-                {
-                    return item.Text;
-                }
-            }
-
-            return value;
-        }
-    }
-}
+    using DotNetNuke.Common.Utilities;
+    using DotNetNuke.Entities.Portals;
+    using DotNetNuke.Entities.Profile;
+    using DotNetNuke.Services.FileSystem;
+
+    /// <summary>
+    /// The UserProfile class provides a Business Layer entity for the Users Profile.
+    /// </summary>
+    [Serializable]
+    public class UserProfile : IIndexable
+    {
+#pragma warning disable SA1310 // Field names should not contain underscore
+#pragma warning disable SA1600 // Elements should be documented
+        // Name properties
+        [Obsolete("Deprecated in 9.8. Scheduled removal in v10.0.0.")]

```

Comment:
What's the suggested replacement if you were using these fields previously?

Reply:
Maybe just say `Deprectated in v9.8.  Use the properties on this class instead. Scheduled removal in v10.0.0`?

Reply:
I have applied the comments suggestion, I do believe these should have never been made public in the first place. I would gladly be overruled, but I see no reason to expose those as we have much easier way to get to those properties.

Reply:
Agreed, those shouldn't need to be exposed.